core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -606,33 +606,33 @@ class SnappyParser(session: SnappySession)
 
   protected final def primary: Rule1[Expression] = rule {
     identifier ~ (
-        '(' ~ ws ~ (
-            '*' ~ ws ~ ')' ~ ws ~> ((udfName: String) =>
-              if (udfName.equalsIgnoreCase("COUNT")) {
-                AggregateExpression(Count(Literal(1, IntegerType)),
-                  mode = Complete, isDistinct = false)
-              } else {
-                throw Utils.analysisException(s"invalid expression $udfName(*)")
-              }) |
-            (DISTINCT ~> trueFn).? ~ (expression * commaSep) ~ ')' ~ ws ~
-                (OVER ~ windowSpec).? ~> { (u: Any, d: Any, e: Any, w: Any) =>
-              val udfName = u.asInstanceOf[String]
-              val exprs = e.asInstanceOf[Seq[Expression]]
-              val function = if (d.asInstanceOf[Option[Boolean]].isEmpty) {
-                UnresolvedFunction(udfName, exprs, isDistinct = false)
-              } else if (udfName.equalsIgnoreCase("COUNT")) {
-                aggregate.Count(exprs).toAggregateExpression(isDistinct = true)
-              } else {
-                UnresolvedFunction(udfName, exprs, isDistinct = true)
-              }
-              w.asInstanceOf[Option[WindowSpec]] match {
-                case None => function
-                case Some(spec: WindowSpecDefinition) =>
-                  WindowExpression(function, spec)
-                case Some(ref: WindowSpecReference) =>
-                  UnresolvedWindowExpression(function, ref)
-              }
+      ('.' ~ identifier).? ~ '(' ~ ws ~ (
```

Comment:
What is this trying to do? An identifier preceeded by a '.'?

The qualified name case is already taken care of below in the '|' case: '.' ~ ws ~ (...) that creates an UnresolvedAttribute. If that does not work for UDFs then add whatever code is required there. Ideally UnresolvedAttribute should get resolved to a UDF during some analysis phase.